packages/web/src/components/apps/header.tsx

```diff
@@ -57,6 +60,13 @@ export default function EditorHeader(props: PropsType) {
           }}
         />
       )}
+
+      {npmInstallStatus === 'installing' ? (
+        <div className="fixed top-0 left-0 right-0 z-[51] h-0.5 pointer-events-none">
+          <div className="h-full w-full bg-white animate-indeterminate" />
+        </div>
+      ) : null}
+
```

Comment:
I added this indeterminate loading state bar thing across the top. I am kinda a fan of it in this case (a global state of some sort making it very clear to the user something is going on). I tried a smaller "chip" type thing in the header bar as well.

Let me know your thoughts. It would not be hard to pull this out if desired.

Reply:
Generally like the idea of having a "status" somewhere in the header for installing packages, loading the devserver, or awaiting an AI response. Will see how this feels but in terms of design but props wise 👍